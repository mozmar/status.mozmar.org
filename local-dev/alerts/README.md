Alerts in this directory will not be put in source control.

Feel free to copy the alerts from `/alerts` here for testing. Or manually create
a few. Do what you feel. It's your (local) world.
