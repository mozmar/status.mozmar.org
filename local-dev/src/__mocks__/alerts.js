const alerts = [
    {
        alertLogged: '2016-12-05-17-23-45-42',
        alertOccurred: '2016-12-05-17-20-12-34',
        description: 'The decline in spirograph and the rise in gang activity',
        graph: 'https://frinkiac.com/img/S07E02/1108189.jpg',
        link: 'https://frinkiac.com/episode/S07E02/1108189',
        resolved: false,
        severity: 'h'
    },
    {
        alertLogged: '2016-12-17-16-53-12-23',
        alertOccurred: '2016-12-17-12-30-12-34',
        description: 'Disco record sales were up 400%',
        graph: 'https://frinkiac.com/img/S08E11/288721.jpg',
        link: 'https://frinkiac.com/episode/S08E11/290589',
        resolved: true,
        severity: 'l'
    },
    {
        alertLogged: '2016-12-14-12-45-12-34',
        alertOccurred: '2016-12-14-12-30-12-34',
        description: 'As intelligence goes up, happiness often goes down',
        graph: 'https://frinkiac.com/img/S12E09/1066440.jpg',
        link: 'https://frinkiac.com/episode/S12E09/1066440',
        resolved: false,
        severity: 'm'
    }
];

export default alerts;
