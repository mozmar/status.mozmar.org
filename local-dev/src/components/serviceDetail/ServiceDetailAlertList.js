import React, { Component } from 'react';

import ServiceDetailAlert from './ServiceDetailAlert';

class ServiceDetailAlertList extends Component {
    static propTypes = {
        alerts: React.PropTypes.array.isRequired
    }

    // Bind 'this' for class functions - otherwise 'this' is undefined
    constructor() {
        super();
        this.renderServiceDetailAlert = this.renderServiceDetailAlert.bind(this);
    }

    renderServiceDetailAlert(index) {
        return (
            <li key={index} className="list-group-item">
                <ServiceDetailAlert alertData={this.props.alerts[index]} />
            </li>
        );
    }

    render() {
        if (this.props.alerts.length > 0) {
            return (
                <div className="service-detail-alerts">
                    <ul className="service-detail-alert-list list-group">
                        {Object.keys(this.props.alerts).map(this.renderServiceDetailAlert)}
                    </ul>
                </div>
            );
        } else {
            // Empty div while data is fetched.
            return (
                <div></div>
            );
        }
    }
}

export default ServiceDetailAlertList;
