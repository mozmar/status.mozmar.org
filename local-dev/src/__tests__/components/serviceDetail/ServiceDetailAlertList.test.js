import React from 'react';
import renderer from 'react-test-renderer';

import ServiceDetailAlertList from '../../../components/serviceDetail/ServiceDetailAlertList';

import alerts from '../../../__mocks__/alerts';

describe('test ServiceDetailAlertList renders', () => {
    test('render ServiceDetailAlertList with no alerts', () => {
        const component = renderer.create(
            <ServiceDetailAlertList alerts={[]} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render ServiceDetailAlertList with alerts', () => {
        const component = renderer.create(
            <ServiceDetailAlertList alerts={alerts} />
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
