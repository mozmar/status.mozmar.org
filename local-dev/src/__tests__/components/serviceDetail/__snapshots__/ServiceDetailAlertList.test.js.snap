exports[`test ServiceDetailAlertList renders render ServiceDetailAlertList with alerts 1`] = `
<div
  className="service-detail-alerts">
  <ul
    className="service-detail-alert-list list-group">
    <li
      className="list-group-item">
      <div
        className="service-detail-alert">
        <h4>
          <a
            href="https://frinkiac.com/episode/S07E02/1108189"
            rel="external">
            The decline in spirograph and the rise in gang activity
          </a>
        </h4>
        <div
          className="service-detail-alert-data-wrapper">
          <dl
            className="service-detail-alert-data-list">
            <dt>
              Resolved
            </dt>
            <dd>
              <span
                className="label label-danger">
                no
              </span>
            </dd>
            <dt>
              Severity
            </dt>
            <dd>
              <span
                className="label label-warning">
                high
              </span>
            </dd>
            <dt>
              Occurred
            </dt>
            <dd>
              Thu Jan 05 2017 11:20:12 GMT-0600 (CST)
            </dd>
            <dt>
              Logged
            </dt>
            <dd>
              Thu Jan 05 2017 11:23:45 GMT-0600 (CST)
            </dd>
          </dl>
          <div
            className="service-detail-alert-graph-wrapper">
            <iframe
              className="service-detail-alert-graph"
              src="https://frinkiac.com/img/S07E02/1108189.jpg" />
          </div>
        </div>
      </div>
    </li>
    <li
      className="list-group-item">
      <div
        className="service-detail-alert">
        <h4>
          <a
            href="https://frinkiac.com/episode/S08E11/290589"
            rel="external">
            Disco record sales were up 400%
          </a>
        </h4>
        <div
          className="service-detail-alert-data-wrapper">
          <dl
            className="service-detail-alert-data-list">
            <dt>
              Resolved
            </dt>
            <dd>
              <span
                className="label label-success">
                yes
              </span>
            </dd>
            <dt>
              Severity
            </dt>
            <dd>
              <span
                className="label label-default">
                low
              </span>
            </dd>
            <dt>
              Occurred
            </dt>
            <dd>
              Tue Jan 17 2017 06:30:12 GMT-0600 (CST)
            </dd>
            <dt>
              Logged
            </dt>
            <dd>
              Tue Jan 17 2017 10:53:12 GMT-0600 (CST)
            </dd>
          </dl>
          <div
            className="service-detail-alert-graph-wrapper">
            <iframe
              className="service-detail-alert-graph"
              src="https://frinkiac.com/img/S08E11/288721.jpg" />
          </div>
        </div>
      </div>
    </li>
    <li
      className="list-group-item">
      <div
        className="service-detail-alert">
        <h4>
          <a
            href="https://frinkiac.com/episode/S12E09/1066440"
            rel="external">
            As intelligence goes up, happiness often goes down
          </a>
        </h4>
        <div
          className="service-detail-alert-data-wrapper">
          <dl
            className="service-detail-alert-data-list">
            <dt>
              Resolved
            </dt>
            <dd>
              <span
                className="label label-danger">
                no
              </span>
            </dd>
            <dt>
              Severity
            </dt>
            <dd>
              <span
                className="label label-info">
                medium
              </span>
            </dd>
            <dt>
              Occurred
            </dt>
            <dd>
              Sat Jan 14 2017 06:30:12 GMT-0600 (CST)
            </dd>
            <dt>
              Logged
            </dt>
            <dd>
              Sat Jan 14 2017 06:45:12 GMT-0600 (CST)
            </dd>
          </dl>
          <div
            className="service-detail-alert-graph-wrapper">
            <iframe
              className="service-detail-alert-graph"
              src="https://frinkiac.com/img/S12E09/1066440.jpg" />
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`test ServiceDetailAlertList renders render ServiceDetailAlertList with no alerts 1`] = `<div />`;
